src/lib/connection/connection.js

```diff
@@ -54,11 +54,13 @@ export const disconnect = () => {
 const sendMessage = async (msg, ...payload) => {
   log.info('connection/sendMessage ' + msg);
   log.debug('connection/sendMessage/payload ' + JSON.stringify(payload));
+
   if (!connector) {
-    connector = await connect();
+    connector = connect();
   }
 
-  return connector(msg, ...payload);
+  let sendFn = await connector;
+  return sendFn(msg, ...payload);
```

Comment:
Actually you could also write directly `await connector(...` but this way it's more readable.

Reply:
I dont think so as await connecetor would just drop the sendFunction. Also connector would be a Promise. Maybe if await connector would change the type of connector internaly from promise to sendFN but i am not quite shure about it.